[CI] TimeSeriesMetricsIT testManySteps failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/gxc2vrx3a5zy6/tests/:server:internalClusterTest/org.elasticsearch.timeseries.support.TimeSeriesMetricsIT/testManySteps

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.timeseries.support.TimeSeriesMetricsIT.testManySteps" -Dtests.seed=7622EAB321665ED6 -Dtests.locale=el -Dtests.timezone=America/Whitehorse -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.timeseries.support.TimeSeriesMetricsIT&tests.test=testManySteps

**Failure excerpt:**
```
java.lang.AssertionError: Expected a map containing
Tuple [v1=v, v2={dim=dim}]: a list containing
                           0: <2021-01-01T00:00:00.000Z=0.4551297387042742>
                           1: <2021-01-01T00:00:04.999Z=0.7485262537259498>
                           2: <2021-01-01T00:00:05.000Z=0.06910621516341653>
                           3: <2021-01-01T00:00:10.000Z=0.5834115216573992>
                           4: <2021-01-01T00:00:14.999Z=0.729380889565715>
                           5: <2021-01-01T00:00:15.000Z=0.23374119894171885>
                           6: <2021-01-01T00:00:19.999Z=0.6322932146116457>
                           7: <2021-01-01T00:00:20.000Z=0.9432465127358071>
                           8: <2021-01-01T00:00:25.000Z=0.6282098545274359>
                           9: <2021-01-01T00:00:30.000Z=0.1986163736396087>
                          10: <2021-01-01T00:00:34.999Z=0.9557792720522522>
                          11: <2021-01-01T00:00:35.000Z=0.9016222067275699>
                          12: <2021-01-01T00:00:40.000Z=0.4699220966529055>
                          13: <2021-01-01T00:00:44.999Z=0.36552271389986024>
                          14: <2021-01-01T00:00:45.000Z=0.4720142800420899>
                          15: <2021-01-01T00:00:49.999Z=0.7546810957114379>
                          16: <2021-01-01T00:00:50.000Z=0.7335687038583637>
                          17: <2021-01-01T00:00:55.000Z=0.6713546718821035>
                          18: <2021-01-01T00:00:59.999Z=0.7483808789801231>
                          19: <2021-01-01T00:01:00.000Z=0.6669291476433954>
                          20: <2021-01-01T00:01:04.999Z=0.5551205052707621>
                          21: <2021-01-01T00:01:05.000Z=0.5705932271106092>
                          22: <2021-01-01T00:01:09.999Z=0.27777033805003004>
                          23: <2021-01-01T00:01:10.000Z=0.4098798336208601>
                          24: <2021-01-01T00:01:14.999Z=0.9564344234899667>
                          25: <2021-01-01T00:01:15.000Z=0.2678063983638509>
                          26: <2021-01-01T00:01:19.999Z=0.44013393690657165>
                          27: <2021-01-01T00:01:20.000Z=0.843895110099188>
                          28: <2021-01-01T00:01:24.999Z=0.6004536424733807>
                          29: <2021-01-01T00:01:25.000Z=0.9574006573914045>
                          30: <2021-01-01T00:01:30.000Z=0.6585716206687732>
                          31: <2021-01-01T00:01:35.000Z=0.8342435965553686>
                          32: <2021-01-01T00:01:40.000Z=0.3933962644210489>
                          33: <2021-01-01T00:01:45.000Z=0.08386849299383348>
                          34: <2021-01-01T00:01:50.000Z=0.3299963902841695>
                          35: <2021-01-01T00:01:54.999Z=0.7239265312521174>
                          36: <2021-01-01T00:01:55.000Z=0.7386963931814903>
                          37: <2021-01-01T00:02:00.000Z=0.943167890739917>
                          38: <2021-01-01T00:02:05.000Z=0.9751943329628867>
                          39: <2021-01-01T00:02:10.000Z=0.3442903413376488>
                          40: <2021-01-01T00:02:15.000Z=0.2604928459833481>
                          41: <2021-01-01T00:02:20.000Z=0.7008197682562844>
                          42: <2021-01-01T00:02:24.999Z=0.09264030093367359>
                          43: <2021-01-01T00:02:25.000Z=0.44202896905148514>
                          44: <2021-01-01T00:02:29.999Z=0.5379917980316546>
                          45: <2021-01-01T00:02:30.000Z=0.14852740069289172>
                          46: <2021-01-01T00:02:35.000Z=0.8914095561698685>
                          47: <2021-01-01T00:02:39.999Z=0.8869424859863377>
                          48: <2021-01-01T00:02:40.000Z=0.08237582990042613>
                          49: <2021-01-01T00:02:45.000Z=0.7871913221420113>
                          50: <2021-01-01T00:02:50.000Z=0.7516884627463615>
                          51: <2021-01-01T00:02:54.999Z=0.46532180295055825>
                          52: <2021-01-01T00:02:55.000Z=0.3542074987717295>
                          53: <2021-01-01T00:03:00.000Z=0.839329469335387>
                          54: <2021-01-01T00:03:04.999Z=0.41727869781814475>
                          55: <2021-01-01T00:03:05.000Z=0.8085243025248715>
                          56: <2021-01-01T00:03:10.000Z=0.007792214236753137>
                          57: <2021-01-01T00:03:14.999Z=0.6761033913571318>
                          58: <2021-01-01T00:03:15.000Z=0.945735852013256>
                          59: <2021-01-01T00:03:20.000Z=0.6473870826444464>
                          60: <2021-01-01T00:03:24.999Z=0.670951490084183>
                          61: <2021-01-01T00:03:25.000Z=0.14639393785400878>
                          62: <2021-01-01T00:03:30.000Z=0.37047250651343944>
                          63: <2021-01-01T00:03:34.999Z=0.9908696227414945>
                          64: <2021-01-01T00:03:35.000Z=0.7742210529481296>
                          65: <2021-01-01T00:03:40.000Z=0.3588504584955219>
                          66: <2021-01-01T00:03:44.999Z=0.6089471902110216>
                          67: <2021-01-01T00:03:45.000Z=0.06432757327048799>
                          68: <2021-01-01T00:03:49.999Z=0.06389508085935103>
                          69: <2021-01-01T00:03:50.000Z=0.2532937785266578>
                          70: <2021-01-01T00:03:55.000Z=0.8249465084877867>
                          71: <2021-01-01T00:04:00.000Z=0.9502552250186133>
                          72: <2021-01-01T00:04:04.999Z=0.555828244719204>
                          73: <2021-01-01T00:04:05.000Z=0.8302026611579609>
                          74: <2021-01-01T00:04:09.999Z=0.5992591966085192>
                          75: <2021-01-01T00:04:10.000Z=0.5657798997164128>
                          76: <2021-01-01T00:04:15.000Z=0.13095347165068572>
                          77: <2021-01-01T00:04:20.000Z=0.12776170539789067>
                          78: <2021-01-01T00:04:24.999Z=0.9793636849452465>
                          79: <2021-01-01T00:04:25.000Z=0.4702729482172464>
                          80: <2021-01-01T00:04:30.000Z=0.6393581725772154>
                          81: <2021-01-01T00:04:34.999Z=0.9687029832655485>
                          82: <2021-01-01T00:04:35.000Z=0.737342205172745>
                          83: <2021-01-01T00:04:40.000Z=0.11258464685492464>
                          84: <2021-01-01T00:04:44.999Z=0.08057494679127608>
                          85: <2021-01-01T00:04:45.000Z=0.20137822561520924>
                          86: <2021-01-01T00:04:50.000Z=0.49241810407926045>
                          87: <2021-01-01T00:04:55.000Z=0.5640491852768988>
                          88: <2021-01-01T00:04:59.999Z=0.06787660126708783>
                          89: <2021-01-01T00:05:00.000Z=0.011973736906484511>
                          90: <2021-01-01T00:05:05.000Z=0.13890758085727084>
                          91: <2021-01-01T00:05:09.999Z=0.6746232013176242>
                          92: <2021-01-01T00:05:10.000Z=0.9298113289427435>
                          93: <2021-01-01T00:05:14.999Z=0.09154615128282007>
                          94: <2021-01-01T00:05:15.000Z=0.5746526609809449>
                          95: <2021-01-01T00:05:19.999Z=0.7889055972806583>
                          96: <2021-01-01T00:05:20.000Z=0.6183892343107337>
                          97: <2021-01-01T00:05:24.999Z=0.38262739162135073>
                          98: <2021-01-01T00:05:25.000Z=0.06019059319295872>
                          99: <2021-01-01T00:05:30.000Z=0.11849337617870181>
                         100: expected <2021-01-01T00:05:34.999Z=0.1698329273975857> but was <missing>
                         101: expected <2021-01-01T00:05:35.000Z=0.8065872669834374> but was <missing>
                         102: expected <2021-01-01T00:05:39.999Z=0.43784662313167777> but was <missing>
                         103: expected <2021-01-01T00:05:40.000Z=0.2075411765126287> but was <missing>
                         104: expected <2021-01-01T00:05:45.000Z=0.024228756254963235> but was <missing>
                         105: expected <2021-01-01T00:05:49.999Z=0.3384234818691708> but was <missing>
                         106: expected <2021-01-01T00:05:50.000Z=0.8489212493663634> but was <missing>
                         107: expected <2021-01-01T00:05:54.999Z=0.9231174129610852> but was <missing>
                         108: expected <2021-01-01T00:05:55.000Z=0.9514452226426685> but was <missing>
                         109: expected <2021-01-01T00:06:00.000Z=0.4617348325843589> but was <missing>
                         110: expected <2021-01-01T00:06:04.999Z=0.04404028735693155> but was <missing>
                         111: expected <2021-01-01T00:06:05.000Z=0.009915152695963547> but was <missing>
                         112: expected <2021-01-01T00:06:10.000Z=0.7286221085982443> but was <missing>
                         113: expected <2021-01-01T00:06:14.999Z=0.010617951443883> but was <missing>
                         114: expected <2021-01-01T00:06:15.000Z=0.07951689141426421> but was <missing>
                         115: expected <2021-01-01T00:06:20.000Z=0.4080373840609848> but was <missing>
                         116: expected <2021-01-01T00:06:24.999Z=0.9142297444114408> but was <missing>
                         117: expected <2021-01-01T00:06:25.000Z=0.1384617485611065> but was <missing>
                         118: expected <2021-01-01T00:06:29.999Z=0.41202144215154113> but was <missing>
                         119: expected <2021-01-01T00:06:30.000Z=0.3678501888248189> but was <missing>
                         120: expected <2021-01-01T00:06:34.999Z=0.7489590107333199> but was <missing>
                         121: expected <2021-01-01T00:06:35.000Z=0.6133704465933951> but was <missing>
                         122: expected <2021-01-01T00:06:40.000Z=0.6377353749661506> but was <missing>
                         123: expected <2021-01-01T00:06:44.999Z=0.6762784134817745> but was <missing>
                         124: expected <2021-01-01T00:06:45.000Z=0.1419206843573424> but was <missing>
                         125: expected <2021-01-01T00:06:50.000Z=0.11726909744952108> but was <missing>
                         126: expected <2021-01-01T00:06:55.000Z=0.4826327945621238> but was <missing>
                         127: expected <2021-01-01T00:07:00.000Z=0.8651886206787438> but was <missing>
                         128: expected <2021-01-01T00:07:04.999Z=0.5792086946309478> but was <missing>
                         129: expected <2021-01-01T00:07:05.000Z=0.8738259886805628> but was <missing>
                         130: expected <2021-01-01T00:07:09.999Z=0.09051287428266563> but was <missing>
                         131: expected <2021-01-01T00:07:10.000Z=0.28634333041764015> but was <missing>
                         132: expected <2021-01-01T00:07:14.999Z=0.7275540462753204> but was <missing>
                         133: expected <2021-01-01T00:07:15.000Z=0.4328669069450125> but was <missing>
                         134: expected <2021-01-01T00:07:19.999Z=0.9913345498059726> but was <missing>
                         135: expected <2021-01-01T00:07:20.000Z=0.9883789342508085> but was <missing>
                         136: expected <2021-01-01T00:07:25.000Z=0.5203523874336384> but was <missing>
                         137: expected <2021-01-01T00:07:30.000Z=0.024588853789757947> but was <missing>
                         138: expected <2021-01-01T00:07:34.999Z=0.7416093450887493> but was <missing>
                         139: expected <2021-01-01T00:07:35.000Z=0.31441479527553073> but was <missing>
                         140: expected <2021-01-01T00:07:40.000Z=0.5375621633944664> but was <missing>
                         141: expected <2021-01-01T00:07:44.999Z=0.108902192886421> but was <missing>
                         142: expected <2021-01-01T00:07:45.000Z=0.7649037414208626> but was <missing>
                         143: expected <2021-01-01T00:07:49.999Z=0.29589582169456197> but was <missing>
                         144: expected <2021-01-01T00:07:50.000Z=0.6541100573801683> but was <missing>
                         145: expected <2021-01-01T00:07:54.999Z=0.042056757103395515> but was <missing>
                         146: expected <2021-01-01T00:07:55.000Z=0.7440128532384114> but was <missing>
                         147: expected <2021-01-01T00:08:00.000Z=0.2826807150000388> but was <missing>
                         148: expected <2021-01-01T00:08:04.999Z=0.22358842375928745> but was <missing>
                         149: expected <2021-01-01T00:08:05.000Z=0.5385074096646784> but was <missing>
                         150: expected <2021-01-01T00:08:09.999Z=0.46280445333336384> but was <missing>
                         151: expected <2021-01-01T00:08:10.000Z=0.9906590696330075> but was <missing>
                         152: expected <2021-01-01T00:08:14.999Z=0.10249288609254781> but was <missing>
                         153: expected <2021-01-01T00:08:15.000Z=0.08642436232063488> but was <missing>
                         154: expected <2021-01-01T00:08:19.999Z=0.9249013422117551> but was <missing>
                         155: expected <2021-01-01T00:08:20.000Z=0.7999287534329088> but was <missing>
                         156: expected <2021-01-01T00:08:24.999Z=0.7348054501358876> but was <missing>
                         157: expected <2021-01-01T00:08:25.000Z=0.38706343984448455> but was <missing>
                         158: expected <2021-01-01T00:08:29.999Z=0.41905785195185674> but was <missing>
                         159: expected <2021-01-01T00:08:30.000Z=0.3375125275834001> but was <missing>
                         160: expected <2021-01-01T00:08:35.000Z=0.2946857970684531> but was <missing>
                         161: expected <2021-01-01T00:08:40.000Z=0.49390342578692625> but was <missing>
                         162: expected <2021-01-01T00:08:45.000Z=0.3735613152874746> but was <missing>
                         163: expected <2021-01-01T00:08:50.000Z=0.5105129968634158> but was <missing>
                         164: expected <2021-01-01T00:08:55.000Z=0.054902492533067115> but was <missing>
                         165: expected <2021-01-01T00:08:59.999Z=0.7041363545281037> but was <missing>
                         166: expected <2021-01-01T00:09:00.000Z=0.36476470363281477> but was <missing>
                         167: expected <2021-01-01T00:09:04.999Z=0.37502150085082775> but was <missing>
                         168: expected <2021-01-01T00:09:05.000Z=0.49262228925717455> but was <missing>
                         169: expected <2021-01-01T00:09:09.999Z=0.23046616975759482> but was <missing>
                         170: expected <2021-01-01T00:09:10.000Z=0.4548015590377461> but was <missing>
                         171: expected <2021-01-01T00:09:15.000Z=0.1848153438932918> but was <missing>
                         172: expected <2021-01-01T00:09:20.000Z=0.8834305280365184> but was <missing>
                         173: expected <2021-01-01T00:09:24.999Z=0.7640370560547635> but was <missing>
                         174: expected <2021-01-01T00:09:25.000Z=0.5721174067614762> but was <missing>
                         175: expected <2021-01-01T00:09:29.999Z=0.46966517000971> but was <missing>
                         176: expected <2021-01-01T00:09:30.000Z=0.9247936809835811> but was <missing>
                         177: expected <2021-01-01T00:09:35.000Z=0.4880043392105796> but was <missing>
                         178: expected <2021-01-01T00:09:39.999Z=0.3141527916170672> but was <missing>
                         179: expected <2021-01-01T00:09:40.000Z=0.6947259329772932> but was <missing>
                         180: expected <2021-01-01T00:09:45.000Z=0.6366530496103533> but was <missing>
                         181: expected <2021-01-01T00:09:50.000Z=0.33653494246137716> but was <missing>
                         182: expected <2021-01-01T00:09:54.999Z=0.06653010344640142> but was <missing>
                         183: expected <2021-01-01T00:09:55.000Z=0.42144838523127914> but was <missing>
                         184: expected <2021-01-01T00:09:59.999Z=0.41733458363095566> but was <missing>
                         185: expected <2021-01-01T00:10:00.000Z=0.11890810970134369> but was <missing>
                         186: expected <2021-01-01T00:10:05.000Z=0.3941475082296084> but was <missing>
                         187: expected <2021-01-01T00:10:09.999Z=0.987797750044224> but was <missing>
                         188: expected <2021-01-01T00:10:10.000Z=0.8135390142527839> but was <missing>
                         189: expected <2021-01-01T00:10:14.999Z=0.9071466820316876> but was <missing>
                         190: expected <2021-01-01T00:10:15.000Z=0.4368907592832023> but was <missing>
                         191: expected <2021-01-01T00:10:20.000Z=0.2676564296677034> but was <missing>
                         192: expected <2021-01-01T00:10:24.999Z=0.3445225038498161> but was <missing>
                         193: expected <2021-01-01T00:10:25.000Z=0.024497239261973225> but was <missing>
                         194: expected <2021-01-01T00:10:29.999Z=0.33978911370526976> but was <missing>
                         195: expected <2021-01-01T00:10:30.000Z=0.860107372478515> but was <missing>
                         196: expected <2021-01-01T00:10:35.000Z=0.12443862493601043> but was <missing>
                         197: expected <2021-01-01T00:10:40.000Z=0.06940172584490356> but was <missing>
                         198: expected <2021-01-01T00:10:44.999Z=0.07996691749120588> but was <missing>
                         199: expected <2021-01-01T00:10:45.000Z=0.2951838714741394> but was <missing>
                         200: expected <2021-01-01T00:10:49.999Z=0.04126816476143602> but was <missing>
                         201: expected <2021-01-01T00:10:50.000Z=0.662594990035955> but was <missing>
                         202: expected <2021-01-01T00:10:55.000Z=0.9798113934945016> but was <missing>
                         203: expected <2021-01-01T00:11:00.000Z=0.8545259035291183> but was <missing>
                         204: expected <2021-01-01T00:11:05.000Z=0.5997512471078215> but was <missing>
                         205: expected <2021-01-01T00:11:10.000Z=0.5451637447697013> but was <missing>
                         206: expected <2021-01-01T00:11:14.999Z=0.0568142010070124> but was <missing>
                         207: expected <2021-01-01T00:11:15.000Z=0.9162508959949868> but was <missing>
                         208: expected <2021-01-01T00:11:19.999Z=0.7446983344288761> but was <missing>
                         209: expected <2021-01-01T00:11:20.000Z=0.5579408149960118> but was <missing>
                         210: expected <2021-01-01T00:11:25.000Z=0.3205206396657406> but was <missing>
                         211: expected <2021-01-01T00:11:29.999Z=0.6974990349319966> but was <missing>
                         212: expected <2021-01-01T00:11:30.000Z=0.040280943588984375> but was <missing>
                         213: expected <2021-01-01T00:11:35.000Z=0.2644033982148998> but was <missing>
                         214: expected <2021-01-01T00:11:39.999Z=0.42222299123726736> but was <missing>
                         215: expected <2021-01-01T00:11:40.000Z=0.2690385808043301> but was <missing>
                         216: expected <2021-01-01T00:11:44.999Z=0.15128608486790762> but was <missing>
                         217: expected <2021-01-01T00:11:45.000Z=0.5171277357545414> but was <missing>
                         218: expected <2021-01-01T00:11:50.000Z=0.03859065433579867> but was <missing>
                         219: expected <2021-01-01T00:11:54.999Z=0.3727845886019916> but was <missing>
                         220: expected <2021-01-01T00:11:55.000Z=0.34206970015803895> but was <missing>
                         221: expected <2021-01-01T00:12:00.000Z=0.9457963759767719> but was <missing>
                         222: expected <2021-01-01T00:12:05.000Z=0.9916651916349725> but was <missing>
                         223: expected <2021-01-01T00:12:10.000Z=0.14894867307824167> but was <missing>
                         224: expected <2021-01-01T00:12:15.000Z=0.9448573955888433> but was <missing>
                         225: expected <2021-01-01T00:12:20.000Z=0.2717494014701203> but was <missing>
                         226: expected <2021-01-01T00:12:24.999Z=0.6590330873996542> but was <missing>
                         227: expected <2021-01-01T00:12:25.000Z=0.3797680349653746> but was <missing>
                         228: expected <2021-01-01T00:12:30.000Z=0.9514213556536654> but was <missing>
                         229: expected <2021-01-01T00:12:34.999Z=0.5306182842698053> but was <missing>
                         230: expected <2021-01-01T00:12:35.000Z=0.5799852130707656> but was <missing>
                         231: expected <2021-01-01T00:12:40.000Z=0.9450223839835813> but was <missing>
                         232: expected <2021-01-01T00:12:45.000Z=0.5660239060342634> but was <missing>
                         233: expected <2021-01-01T00:12:49.999Z=0.38579889609186035> but was <missing>
                         234: expected <2021-01-01T00:12:50.000Z=0.6206251138359106> but was <missing>
                         235: expected <2021-01-01T00:12:54.999Z=0.8783859884375864> but was <missing>
                         236: expected <2021-01-01T00:12:55.000Z=0.41562477947925125> but was <missing>
                         237: expected <2021-01-01T00:13:00.000Z=0.49497384696690583> but was <missing>
                         238: expected <2021-01-01T00:13:04.999Z=0.8368596737081982> but was <missing>
                         239: expected <2021-01-01T00:13:05.000Z=0.026848906334563938> but was <missing>
                         240: expected <2021-01-01T00:13:10.000Z=0.3215280778417898> but was <missing>
                         241: expected <2021-01-01T00:13:14.999Z=0.19335953839300524> but was <missing>
                         242: expected <2021-01-01T00:13:15.000Z=0.6763431240759723> but was <missing>
                         243: expected <2021-01-01T00:13:19.999Z=0.31101540275950057> but was <missing>
                         244: expected <2021-01-01T00:13:20.000Z=0.35875680973166857> but was <missing>
                         245: expected <2021-01-01T00:13:24.999Z=0.6995294125062822> but was <missing>
                         246: expected <2021-01-01T00:13:25.000Z=0.8298776213536803> but was <missing>
                         247: expected <2021-01-01T00:13:29.999Z=0.6558258984605987> but was <missing>
                         248: expected <2021-01-01T00:13:30.000Z=0.8082640715272454> but was <missing>
                         249: expected <2021-01-01T00:13:34.999Z=0.6830623546694924> but was <missing>
                         250: expected <2021-01-01T00:13:35.000Z=0.08667542512164306> but was <missing>
                         251: expected <2021-01-01T00:13:39.999Z=0.19729407621348383> but was <missing>
                         252: expected <2021-01-01T00:13:40.000Z=0.6932908173948499> but was <missing>
                         253: expected <2021-01-01T00:13:44.999Z=0.525526254359483> but was <missing>
                         254: expected <2021-01-01T00:13:45.000Z=0.6427289496963308> but was <missing>
                         255: expected <2021-01-01T00:13:50.000Z=0.5869770443795856> but was <missing>
                         256: expected <2021-01-01T00:13:54.999Z=0.5674164887746508> but was <missing>
                         257: expected <2021-01-01T00:13:55.000Z=0.6472085992432111> but was <missing>
                         258: expected <2021-01-01T00:14:00.000Z=0.5080826372822587> but was <missing>
                         259: expected <2021-01-01T00:14:04.999Z=0.9597422293350149> but was <missing>
                         260: expected <2021-01-01T00:14:05.000Z=0.9270671081988116> but was <missing>
                         261: expected <2021-01-01T00:14:09.999Z=0.12248739006365983> but was <missing>
                         262: expected <2021-01-01T00:14:10.000Z=0.7702236638751139> but was <missing>
                         263: expected <2021-01-01T00:14:15.000Z=0.7087306361100165> but was <missing>
                         264: expected <2021-01-01T00:14:20.000Z=0.6843871707457364> but was <missing>
                         265: expected <2021-01-01T00:14:25.000Z=0.17830916514380546> but was <missing>
                         266: expected <2021-01-01T00:14:29.999Z=0.6210697829553862> but was <missing>
                         267: expected <2021-01-01T00:14:30.000Z=0.11952053278007113> but was <missing>
                         268: expected <2021-01-01T00:14:35.000Z=0.4907080695735808> but was <missing>
                         269: expected <2021-01-01T00:14:39.999Z=0.5044229329899617> but was <missing>
                         270: expected <2021-01-01T00:14:40.000Z=0.8098449576363388> but was <missing>
                         271: expected <2021-01-01T00:14:45.000Z=0.7073947732309276> but was <missing>
                         272: expected <2021-01-01T00:14:50.000Z=0.741745360604035> but was <missing>
                         273: expected <2021-01-01T00:14:54.999Z=0.045642491936204066> but was <missing>
                         274: expected <2021-01-01T00:14:55.000Z=0.3925525054875817> but was <missing>
                         275: expected <2021-01-01T00:14:59.999Z=0.5611557283173259> but was <missing>
                         276: expected <2021-01-01T00:15:00.000Z=0.35400857091463045> but was <missing>
                         277: expected <2021-01-01T00:15:05.000Z=0.09168093525373877> but was <missing>
                         278: expected <2021-01-01T00:15:10.000Z=0.5475707225973417> but was <missing>
                         279: expected <2021-01-01T00:15:14.999Z=0.664752762568858> but was <missing>
                         280: expected <2021-01-01T00:15:15.000Z=0.08912724907486924> but was <missing>
                         281: expected <2021-01-01T00:15:20.000Z=0.6585387850803934> but was <missing>
                         282: expected <2021-01-01T00:15:24.999Z=0.09029273838340901> but was <missing>
                         283: expected <2021-01-01T00:15:25.000Z=0.409504420014975> but was <missing>
                         284: expected <2021-01-01T00:15:29.999Z=0.5651749841978154> but was <missing>
                         285: expected <2021-01-01T00:15:30.000Z=0.8878301126587491> but was <missing>
                         286: expected <2021-01-01T00:15:34.999Z=0.9281937114590947> but was <missing>
                         287: expected <2021-01-01T00:15:35.000Z=0.06646301838087065> but was <missing>
                         288: expected <2021-01-01T00:15:39.999Z=0.4735243111267289> but was <missing>
                         289: expected <2021-01-01T00:15:40.000Z=0.16506436612446573> but was <missing>
                         290: expected <2021-01-01T00:15:44.999Z=0.22065778611663822> but was <missing>
                         291: expected <2021-01-01T00:15:45.000Z=0.9905637641344318> but was <missing>
                         292: expected <2021-01-01T00:15:50.000Z=0.7850213935453285> but was <missing>
                         293: expected <2021-01-01T00:15:54.999Z=0.21579615803550412> but was <missing>
                         294: expected <2021-01-01T00:15:55.000Z=0.304878903442504> but was <missing>
                         295: expected <2021-01-01T00:16:00.000Z=0.015409635034005387> but was <missing>
                         296: expected <2021-01-01T00:16:05.000Z=0.725866088480892> but was <missing>
                         297: expected <2021-01-01T00:16:10.000Z=0.11254146898894912> but was <missing>
                         298: expected <2021-01-01T00:16:15.000Z=0.5062509646668806> but was <missing>
                         299: expected <2021-01-01T00:16:20.000Z=0.3459109886548972> but was <missing>
                         300: expected <2021-01-01T00:16:24.999Z=0.06227854455431503> but was <missing>
                         301: expected <2021-01-01T00:16:25.000Z=0.4499948775529975> but was <missing>
                         302: expected <2021-01-01T00:16:30.000Z=0.4466990295074956> but was <missing>
                         303: expected <2021-01-01T00:16:35.000Z=0.6403157892915267> but was <missing>
                         304: expected <2021-01-01T00:16:40.000Z=0.35460811726538977> but was <missing>
                         305: expected <2021-01-01T00:16:45.000Z=0.659061881405667> but was <missing>
                         306: expected <2021-01-01T00:16:49.999Z=0.6791216949299511> but was <missing>
                         307: expected <2021-01-01T00:16:50.000Z=0.0039387171400246945> but was <missing>
                         308: expected <2021-01-01T00:16:54.999Z=0.6022733938969255> but was <missing>
                         309: expected <2021-01-01T00:16:55.000Z=0.18676136868684134> but was <missing>
                         310: expected <2021-01-01T00:17:00.000Z=0.2666263570400442> but was <missing>
                         311: expected <2021-01-01T00:17:04.999Z=0.15643966865189407> but was <missing>
                         312: expected <2021-01-01T00:17:05.000Z=0.199943621765991> but was <missing>
                         313: expected <2021-01-01T00:17:10.000Z=0.43746268234011776> but was <missing>
                         314: expected <2021-01-01T00:17:15.000Z=0.7723177878078241> but was <missing>
                         315: expected <2021-01-01T00:17:19.999Z=0.41763277213436567> but was <missing>
                         316: expected <2021-01-01T00:17:20.000Z=0.48210455280317754> but was <missing>
                         317: expected <2021-01-01T00:17:24.999Z=0.4511453528141526> but was <missing>
                         318: expected <2021-01-01T00:17:25.000Z=0.9130225236967551> but was <missing>
                         319: expected <2021-01-01T00:17:30.000Z=0.6210789011495227> but was <missing>
                         320: expected <2021-01-01T00:17:34.999Z=0.22924523568544397> but was <missing>
                         321: expected <2021-01-01T00:17:35.000Z=0.16399779965098615> but was <missing>
                         322: expected <2021-01-01T00:17:40.000Z=0.48744322691080544> but was <missing>
                         323: expected <2021-01-01T00:17:45.000Z=0.26610831973799576> but was <missing>
                         324: expected <2021-01-01T00:17:50.000Z=0.4422157041900304> but was <missing>
                         325: expected <2021-01-01T00:17:54.999Z=0.6860169353820335> but was <missing>
                         326: expected <2021-01-01T00:17:55.000Z=0.7553872377043779> but was <missing>
                         327: expected <2021-01-01T00:17:59.999Z=0.2480655219381852> but was <missing>
                         328: expected <2021-01-01T00:18:00.000Z=0.7213040825749619> but was <missing>
                         329: expected <2021-01-01T00:18:05.000Z=0.9729698054392547> but was <missing>
                         330: expected <2021-01-01T00:18:10.000Z=0.4208461554192252> but was <missing>
                         331: expected <2021-01-01T00:18:15.000Z=0.30416626182711204> but was <missing>
                         332: expected <2021-01-01T00:18:19.999Z=0.07749791000291806> but was <missing>
                         333: expected <2021-01-01T00:18:20.000Z=0.4757073924452886> but was <missing>
                         334: expected <2021-01-01T00:18:25.000Z=0.08554592036336717> but was <missing>
                         335: expected <2021-01-01T00:18:29.999Z=0.7340012583287395> but was <missing>
                         336: expected <2021-01-01T00:18:30.000Z=0.9941635331663973> but was <missing>
                         337: expected <2021-01-01T00:18:34.999Z=0.7828316503597539> but was <missing>
                         338: expected <2021-01-01T00:18:35.000Z=0.30945351301952173> but was <missing>
                         339: expected <2021-01-01T00:18:40.000Z=0.30234854049226056> but was <missing>
                         340: expected <2021-01-01T00:18:45.000Z=0.4002144176760304> but was <missing>
                         341: expected <2021-01-01T00:18:50.000Z=0.32881899569170037> but was <missing>
                         342: expected <2021-01-01T00:18:54.999Z=0.0614987375019721> but was <missing>
                         343: expected <2021-01-01T00:18:55.000Z=0.5084742831743445> but was <missing>
                         344: expected <2021-01-01T00:19:00.000Z=0.8934453952345911> but was <missing>
                         345: expected <2021-01-01T00:19:05.000Z=0.6092914178452781> but was <missing>
                         346: expected <2021-01-01T00:19:09.999Z=0.7004464908699153> but was <missing>
                         347: expected <2021-01-01T00:19:10.000Z=0.6009157811712023> but was <missing>
                         348: expected <2021-01-01T00:19:14.999Z=0.6895813253251787> but was <missing>
                         349: expected <2021-01-01T00:19:15.000Z=0.6239452397065652> but was <missing>
                         350: expected <2021-01-01T00:19:19.999Z=0.6810945577345899> but was <missing>
                         351: expected <2021-01-01T00:19:20.000Z=0.5730112791258264> but was <missing>
                         352: expected <2021-01-01T00:19:24.999Z=0.86800660804153> but was <missing>
                         353: expected <2021-01-01T00:19:25.000Z=0.9733871933514536> but was <missing>
                         354: expected <2021-01-01T00:19:30.000Z=0.6906965554241613> but was <missing>
                         355: expected <2021-01-01T00:19:35.000Z=0.5739682642911249> but was <missing>
                         356: expected <2021-01-01T00:19:39.999Z=0.8990359655937489> but was <missing>
                         357: expected <2021-01-01T00:19:40.000Z=0.09049997981422186> but was <missing>
                         358: expected <2021-01-01T00:19:45.000Z=0.4324231780839509> but was <missing>
                         359: expected <2021-01-01T00:19:49.999Z=0.616003346497332> but was <missing>
                         360: expected <2021-01-01T00:19:50.000Z=0.36203941965651454> but was <missing>
                         361: expected <2021-01-01T00:19:54.999Z=0.17797293934401093> but was <missing>
                         362: expected <2021-01-01T00:19:55.000Z=0.3331551834595773> but was <missing>
                         363: expected <2021-01-01T00:19:59.999Z=0.09035570601766152> but was <missing>
                         364: expected <2021-01-01T00:20:00.000Z=0.7894560954769962> but was <missing>
                         365: expected <2021-01-01T00:20:05.000Z=0.448010283688529> but was <missing>
                         366: expected <2021-01-01T00:20:10.000Z=0.3663577416105226> but was <missing>
                         367: expected <2021-01-01T00:20:14.999Z=0.7276265964764296> but was <missing>
                         368: expected <2021-01-01T00:20:15.000Z=0.6362605716405826> but was <missing>
                         369: expected <2021-01-01T00:20:19.999Z=0.9409268279747252> but was <missing>
                         370: expected <2021-01-01T00:20:20.000Z=0.8583165107949413> but was <missing>
                         371: expected <2021-01-01T00:20:24.999Z=0.7387645499111701> but was <missing>
                         372: expected <2021-01-01T00:20:25.000Z=0.07331642188557785> but was <missing>
                         373: expected <2021-01-01T00:20:30.000Z=0.4903891638382155> but was <missing>
                         374: expected <2021-01-01T00:20:34.999Z=0.3102301543178899> but was <missing>
                         375: expected <2021-01-01T00:20:35.000Z=0.5557337619144058> but was <missing>
                         376: expected <2021-01-01T00:20:40.000Z=0.0717768117526818> but was <missing>
                         377: expected <2021-01-01T00:20:44.999Z=0.09273677162106087> but was <missing>
                         378: expected <2021-01-01T00:20:45.000Z=0.45428670065818666> but was <missing>
                         379: expected <2021-01-01T00:20:50.000Z=0.19948240702096132> but was <missing>
                         380: expected <2021-01-01T00:20:54.999Z=0.7564090715516368> but was <missing>
                         381: expected <2021-01-01T00:20:55.000Z=0.23429707096863506> but was <missing>
                         382: expected <2021-01-01T00:21:00.000Z=0.7201046584443479> but was <missing>
                         383: expected <2021-01-01T00:21:05.000Z=0.3908252868120208> but was <missing>
                         384: expected <2021-01-01T00:21:10.000Z=0.8866702504069978> but was <missing>
                         385: expected <2021-01-01T00:21:15.000Z=0.016825491750226407> but was <missing>
                         386: expected <2021-01-01T00:21:19.999Z=0.5138063909477396> but was <missing>
                         387: expected <2021-01-01T00:21:20.000Z=0.9546463428765657> but was <missing>
                         388: expected <2021-01-01T00:21:25.000Z=0.9299669224098617> but was <missing>
                         389: expected <2021-01-01T00:21:30.000Z=0.6313641340645438> but was <missing>
                         390: expected <2021-01-01T00:21:34.999Z=0.1026536182083384> but was <missing>
                         391: expected <2021-01-01T00:21:35.000Z=0.4248279931224037> but was <missing>
                         392: expected <2021-01-01T00:21:40.000Z=0.5034890447097179> but was <missing>
                         393: expected <2021-01-01T00:21:44.999Z=0.37704655179166> but was <missing>
                         394: expected <2021-01-01T00:21:45.000Z=0.9462089840695226> but was <missing>
                         395: expected <2021-01-01T00:21:49.999Z=0.545432001691098> but was <missing>
                         396: expected <2021-01-01T00:21:50.000Z=0.1340620081786419> but was <missing>
                         397: expected <2021-01-01T00:21:55.000Z=0.25123413355136825> but was <missing>
                         398: expected <2021-01-01T00:22:00.000Z=0.778774116909822> but was <missing>
                         399: expected <2021-01-01T00:22:05.000Z=0.12786778524181597> but was <missing>
                         400: expected <2021-01-01T00:22:10.000Z=0.36516830444724113> but was <missing>
                         401: expected <2021-01-01T00:22:14.999Z=0.7519529722143993> but was <missing>
                         402: expected <2021-01-01T00:22:15.000Z=0.08778373852526666> but was <missing>
                         403: expected <2021-01-01T00:22:20.000Z=0.6077578745815715> but was <missing>
                         404: expected <2021-01-01T00:22:25.000Z=0.886109568533835> but was <missing>
                         405: expected <2021-01-01T00:22:29.999Z=0.1050054631672448> but was <missing>
                         406: expected <2021-01-01T00:22:30.000Z=0.20796312143292472> but was <missing>
                         407: expected <2021-01-01T00:22:35.000Z=0.8153058065096401> but was <missing>

  at __randomizedtesting.SeedInfo.seed([7622EAB321665ED6:17A2B91D02A5B6D7]:0)
  at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:79)
  at io.github.nik9000.mapmatcher.MapMatcher.assertMap(MapMatcher.java:62)
  at org.elasticsearch.timeseries.support.TimeSeriesMetricsIT.testManySteps(TimeSeriesMetricsIT.java:346)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/timeseries/support/TimeSeriesMetricsIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/timeseries/support/TimeSeriesMetricsIT.java
@@ -428,7 +428,7 @@ public class TimeSeriesMetricsIT extends ESIntegTestCase {
     ) {
         return withMetrics(
             bucketBatchSize,
-            between(0, 10000),  // Not used by this method
+            between(1, 10000),
             staleness,
             (future, metrics) -> metrics.range(List.of(eq("v")), List.of(), timeMillis, range, step, new CollectingListener(future))
         );
@@ -480,7 +480,7 @@ public class TimeSeriesMetricsIT extends ESIntegTestCase {
         new TimeSeriesMetricsService(client(), bucketBatchSize, docBatchSize, staleness).newMetrics(
             new String[] { "tsdb" },
             IndicesOptions.STRICT_EXPAND_OPEN,
-            new ActionListener<TimeSeriesMetrics>() {
+            new ActionListener<>() {
                 @Override
                 public void onResponse(TimeSeriesMetrics metrics) {
                     handle.accept(result, metrics);
@@ -511,7 +511,7 @@ public class TimeSeriesMetricsIT extends ESIntegTestCase {
                 results.put(currentDimensions, currentValues);
             }
             currentDimensions = new Tuple<>(metric, dimensions);
-            currentValues = new ArrayList<>();
+            currentValues = results.getOrDefault(currentDimensions, new ArrayList<>());
         }
 
         @Override

